be/src/common/config.h

```diff
@@ -915,5 +915,6 @@ CONF_String(rocksdb_cf_options_string, "block_based_table_factory={block_cache=1
 
 // limit local exchange buffer's memory size per driver
 CONF_Int64(local_exchange_buffer_mem_limit_per_driver, "134217728"); // 128MB
+CONF_mInt64(wait_apply_time, "6000")                                 // 6s
```

Comment:
It means the query may stuck for six seconds?
If the bug fix is not critical for users, I think we can better to design a better solution.

Reply:
> It means the query may stuck for six seconds? If the bug fix is not critical for users, I think we can better to design a better solution.

If there are idle threads, publish version thread will wait up to 6s. If all threads are running, we don't wait and return immediately just like before.